lib/Service/RowService.php

```diff
@@ -435,8 +435,7 @@ public function delete(int $id, ?int $viewId, string $userId): Row2 {
 				$this->logger->error($e->getMessage(), ['exception' => $e]);
 				throw new InternalError(get_class($this) . ' - ' . __FUNCTION__ . ': '.$e->getMessage());
 			}
-			$rowIds = $this->mapper->getRowIdsOfView($view, $userId);
-			if(!in_array($id, $rowIds)) {
+			if(!$this->row2Mapper->isRowInViewPresent($id, $view, $userId)) {
```

Comment:
This seems to be the last usage of $this->mapper so we could drop it from the constructor which makes it one less dependency of the class to resolve

Reply:
@enjeck Could you check that? Then we can merge this :)